src/test/java/org/apache/commons/codec/binary/HexTest.java

```diff
@@ -227,18 +227,19 @@ public void testDecodeByteArrayOddCharacters() {
 
     @Test
     public void testDecodeByteBufferEmpty() throws DecoderException {
-        assertTrue(Arrays.equals(new byte[0], new Hex().decode(ByteBuffer.allocate(0))));
```

Comment:
I would prefer to add tests instead of updating each call to `allocate` with a call to `allocateDirect`. This is something that could be done either with subclasses (one for allocate, one for allocateDirect) or with parameterizing the test with lambdas.

Reply:
I'm sorry but have no time for that